.github/workflows/build-image.yml

```diff
@@ -0,0 +1,66 @@
+# This workflow is intended to be called manually when a new Docker image is needed.
+name: Build and Publish Docker Image
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  build:
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        build: [
+          {name: ubuntu-22.04},
+          {name: ubuntu-20.04},
+        ]
+
+    name: Building docker image ${{ matrix.build.name }}
+    runs-on: ${{ matrix.build.name }}
+
+    steps:
+      - name: Set environment variable
+        run: echo "CI_IMAGE_NAME=ghcr.io/${{ github.repository }}/tt-umd-ci-${{ matrix.build.name }}" >> $GITHUB_ENV
+
+      - name: Fix permissions
+        run: sudo chmod 777 -R $GITHUB_WORKSPACE
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Build image
+
+      - name: Build and export base Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: .github
+          file: .github/${{ matrix.build.name }}.Dockerfile
+          push: true
+          build-args: |
+            GIT_SHA=${{ github.sha }}
+          tags: |
+            ${{ env.CI_IMAGE_NAME}}:${{ github.sha }}
+
```

Comment:
If you have just one image you can set both tags at once. This was split in another project because it was creating multiple images, and I wanted to make sure they could all be built before setting the latest tag.

```
tags: |
            ${{ env.CI_IMAGE_NAME}}:${{ github.sha }}
            ${{ env.CI_IMAGE_NAME}}:latest
```